_episodes/requirements.md

```diff
@@ -18,4 +18,91 @@ By the end of this episode,
 learners should be able to
 __describe all the requirements for running a tool__
 
+
+> ## Exercise 1:
+>
+> For each of these variables, is it required in a CommandLineTool?
+>
+> A. inputs
+> B. outputs
+> C. arguments
+> D. baseCommand
+> E. requirements
+> F. hints
+> > ## Solution
+> > Just like a normal shell script, you need inputs, outputs, and a command to run. So A, B, C are definitely needed. D is sometimes needed, this is where you put the binary invocation of the tool. E and F are necessary for adding environment help, such as machine and docker requirements.
+> {: .solution}
+{: .challenge}
+
+> ## Exercise 2:
+> 
+> When working in a cloud environment, you need to specify what machine type you would like to run on. Which means the job has to have specific RAM, Cores and Disk space parameters.
+>
+> What ResourceRequirements can you specify in a CommandLineTool to satisfy the machine requirements?
+>
+> > ## Solution:
+> >
+> > coresMin and coresMax
+> > ramMin and ramMax
+> > tmpdirMin and tmpdirMax
+> > outdirMin andoutdirMax
+> {: .solution}
+{: .challenge}
+
+> ## Exercise 3:
+>
+> Input arguments for a command in a shell script are typically strings that can be classified as file paths, booleans, integers, etc. When you have multiple inputs of a same type, you classify that as an array.
+>
+> What input types would you use for these inputs?
```

Comment:
I'd recommend to adjust this list to a table, with columns for each possible answer, e.g.

```
input | File | String | Boolean | array of Files | array of Strings |
-----|-----|-----|-----|-----|
BAM |     |     |     |     |
VCF |     |     |     |     |
Sample name |     |     |     |     |
23 Chromosome IDs |     |     |     |     |
Flag for turning on paired-end mode in baw mem |     |     |     |     |
Cohort of sample paired-end FASTQs |    |     |     |     |
```

I also think it might be good to stress in the exercise introduction that special types exist for file paths and booleans: teh way it's written right now I think a learner might make the mistake of saying that all of A, B, C, and E are just strings.